Sources/PenguinParallel/NonblockingThreadPool/NumberOperations.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Returns an array of all positive integers that are co-prime with `n`.
+///
+/// Two numbers are co-prime if their GCD is 1.
+internal func makeCoprimes(upTo n: Int) -> [Int] {
```

Comment:
GCD-related stuff is a great opportunity for generic programming.  See any of Stepanov's talks or writings on "Greatest Common Measure." :-)

API-wise,
```suggestion
internal func positiveCoprimes(_ n: Int) -> [Int] {
```

The `make` prefix is reserved for functions where the distinctness of the return value is significant, e.g. `makeIterator()`.  

But then, thinking this all the way through, step 1 is really

```swift
extension Integer {
   var positiveCoprimes: [Self] { ... }
}
```

and step 2 is

```swift
extension Integer {
   var positiveCoprimes: PositiveCoprimes<Self> { ... }
}
```

which doesn't require any dynamic storage.

Reply:
Yes, I will do this in a follow-up PR. Great suggestion! (I will likely move this over to `PenguinStructures` as that's probably the best spot to put this for now (unless you have a better suggestion)?

Reply:
`PenguinNumerics`?

And BTW, can we drop `Penguin` from the directory names please? ;-)